src/main/scala/com/fulcrumgenomics/umi/FilterConsensusReads.scala

```diff
@@ -0,0 +1,217 @@
+      |being used on reads marked as being mapped to the negative strand.  This is necessary if
+      |the reads have been mapped and the bases/quals reversed but the consensus tags have not.
+      |If true, the tags written to the output BAM will be reversed where necesary in order to
+      |line up with the bases and quals.
+    """
+)
+class FilterConsensusReads
+( @arg(flag="i", doc="The input SAM or BAM file of consensus reads.") val input: PathToBam,
+  @arg(flag="o", doc="Output SAM or BAM file.") val output: PathToBam,
+  @arg(flag="r", doc="Reference fasta file.") val ref: PathToFasta,
+  @arg(flag="R", doc="Reverse [complement] per base tags on reverse strand reads.") val reversePerBaseTags: Boolean = true,
+  @arg(flag="N", doc="Mask (make 'N') consensus bases with quality less than this threshold.") val minBaseQuality: PhredScore = DefaultMinConsensusBaseQuality,
+  @arg(flag="M", doc="The minimum number of reads supporting a consensus base/read.") val minReads: Int = DefaultMinReads,
+  @arg(flag="E", doc="The maximum raw-read error rate across the entire consensus read.") val maxReadErrorRate: Double = 0.025,
+  @arg(flag="e", doc="The maximum error rate for a single consensus base.") val maxBaseErrorRate: Double = 0.1,
+  @arg(flag="n", doc="Maximum fraction of no-calls in the read after filtering.") val maxNoCallFraction: Double = 0.2
+) extends FgBioTool with LazyLogging {
+  // Baseline input validation
+  Io.assertReadable(input)
+  Io.assertReadable(ref)
+  Io.assertCanWriteFile(output)
+  if (maxReadErrorRate < 0 && maxReadErrorRate > 1) fail("max-read-error-rate must be between 0 and 1.")
+  if (maxBaseErrorRate < 0 && maxBaseErrorRate > 1) fail("max-base-error-rate must be between 0 and 1.")
+  if (maxNoCallFraction < 0 && maxNoCallFraction > 1) fail("max-no-call-fraction must be between 0 and 1.")
+
+  // Variables for tracking how many reads meet which fate
+  private var totalReads:  Long = 0
+  private var maskedBases: Long = 0
+  private var totalBases:  Long = 0
+
+  // Case class to track results of filtering a single read
+  private[umi] case class FilterResult(keepRead: Boolean, maskedBases: Int)
+
+  override def execute(): Unit = {
+    val in       = SamReaderFactory.make().open(input)
+    val header   = in.getFileHeader.clone()
+    header.setSortOrder(SortOrder.coordinate)
+    val sorter    = Bams.sortingCollection(SortOrder.coordinate, header)
+    val out       = new SAMFileWriterFactory().setCreateIndex(true).makeWriter(header, true, output.toFile, null)
+    val progress1 = new ProgressLogger(logger, verb="Filtered and masked")
+
+    // Go through the reads by template and do the filtering
+    Bams.templateIterator(in).foreach { template =>
+      val r1 = template.r1.getOrElse(throw new IllegalStateException(s"${template.name} had no R1."))
+      if (r1.getReadPairedFlag) require(template.r2.isDefined, s"Paired read missing R2: ${template.name}")
+      totalReads += template.readCount
```

Comment:
My vote would be for `totalReads`, `totalBases`, and `maskedBases` to only count primary consensus reads.